pallets/liquid-staking-two/src/lib.rs

```diff
@@ -0,0 +1,820 @@
+                    amount,
+                    block_number: frame_system::Pallet::<T>::block_number(),
+                    status: ResponseStatus::Successed,
+                },
+            );
+
+            Self::deposit_event(Event::<T>::RewardsRecorded(agent, amount));
+            Ok(().into())
+        }
+
+        /// Record slash event from relaychain.
+        ///
+        /// Can only invoked by stake client. Decrease asset amount in `StakingPool` and change
+        /// exchange rate.
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn record_slash(
+            origin: OriginFor<T>,
+            agent: T::AccountId,
+            era_index: EraIndex,
+            #[pallet::compact] amount: Balance,
+        ) -> DispatchResultWithPostInfo {
+            // T::WithdrawOrigin::ensure_origin(origin)?;
+            let _who = ensure_signed(origin)?;
+            // ensure!(T::Members::contains(&agent), Error::<T>::IllegalAgent);
+            StakingPool::<T>::try_mutate(|m| -> DispatchResult {
+                *m = m.checked_sub(amount).ok_or(ArithmeticError::Underflow)?;
+                Ok(())
+            })?;
+
+            let exchange_rate = Rate::checked_from_rational(
+                StakingPool::<T>::get(),
+                T::Currency::total_issuance(T::LiquidCurrency::get()),
+            )
+            .ok_or(Error::<T>::InvalidExchangeRate)?;
+            ExchangeRate::<T>::put(exchange_rate);
+
+            StakingOperationHistory::<T>::insert(
+                era_index,
+                StakingOperationType::RecordSlash,
+                Operation::<_> {
+                    amount,
+                    block_number: frame_system::Pallet::<T>::block_number(),
+                    status: ResponseStatus::Successed,
+                },
+            );
+
+            Self::deposit_event(Event::<T>::SlashRecorded(agent, amount));
+            Ok(().into())
+        }
+
+        // bond/unbond/rebond/bond_extra may be merge into one
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn record_bond_response(
+            origin: OriginFor<T>,
+            era_index: Option<EraIndex>,
+        ) -> DispatchResultWithPostInfo {
+            let _who = ensure_signed(origin)?;
+            let era_index = era_index.unwrap_or(Self::current_era());
+
+            let op = StakingOperationHistory::<T>::try_mutate(
+                era_index,
+                StakingOperationType::Bond,
+                |op| -> Result<Operation<_>, DispatchError> {
+                    let next_op = op
+                        .clone()
+                        .filter(|op| op.status == ResponseStatus::Ready)
+                        .map(|op| Operation {
+                            status: ResponseStatus::Successed,
+                            ..op
+                        })
+                        .ok_or(Error::<T>::OperationNotReady)?;
+                    *op = Some(next_op.clone());
```

Comment:
if we only need `amout`, we should remove this `clone` and return back only `amount`